### PR TITLE
Update documentation for single sign key management

### PR DESCRIPTION
**Pull request by lsubash**
- Updated SBL key management and Key ids
- Info with GenerateKeys.py
- Build changes for introducting SBL_KEY_DIR env variable
- GenContainer updates for auth types and Key id

Signed-off-by: Subash Lakkimsetti <subash.lakkimsetti@intel.com>

### CODE REVIEW

**Comment by lsubash** [comment]
Adding @sagraw2 @mauricema @rprangar for review

File: source/developer-guides/build-system.rst [hunk 1]
```diff
@@ -33,6 +33,25 @@ ConfigDataBlob.h             C file for *Internal* configuration data with defau
 See :ref:`customize-build` for more details.
 
 
+.. _sbl-keys:
```
Comment (rprangar) [inline source/developer-guides/build-system.rst:36]: Is this the right place for this info? I think build-host-setup.rst maybe better?

This section talks about the EDK2 build infrastructure.

File: source/developer-guides/build-system.rst [hunk 2]
```diff
@@ -33,6 +33,25 @@ ConfigDataBlob.h             C file for *Internal* configuration data with defau
...
+|SPN| Keys Generation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Generate |SPN| test keys and this is pre-requiste before |SPN| build:
+
+Set environment variable for Sbl Keys directory::
+
+  $set SBL_KEY_DIR=$(SBL_ROOT)\..\SblKeys\
+
+Generate SBLKeys::
```
Comment (rprangar) [inline source/developer-guides/build-system.rst:47]: space between SBL and Keys

File: source/developer-guides/build-system.rst [hunk 3]
```diff
@@ -33,6 +33,25 @@ ConfigDataBlob.h             C file for *Internal* configuration data with defau
 See :ref:`customize-build` for more details.
 
 
+.. _sbl-keys:
+
+|SPN| Keys Generation
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Generate |SPN| test keys and this is pre-requiste before |SPN| build:
```
Comment (rprangar) [inline source/developer-guides/build-system.rst:41]: Why mention test keys?

File: source/security/key-management.rst [hunk 4]
```diff
@@ -39,3 +39,63 @@ Cryptographic keys used for signing |SPN| binaries should be securely managed.
 A leak of private key would allow a hacker to install/push compromised |SPN| or rootkits on platforms.
 Here is a `link <https://csrc.nist.gov/CSRC/media/Publications/white-paper/2018/01/26/security-considerations-for-code-signing/final/documents/security-considerations-for-code-signing.pdf>`_ to NIST whitepaper which provides BKM's for Code signing and secure key management.
 
+
```
Comment (rprangar) [inline source/security/key-management.rst:42]: Can you add a couple of lines providing the context at the top of this page?

File: source/developer-guides/build-system.rst (hunk 1 above)
Comment (rprangar) [reply source/developer-guides/build-system.rst:36]: Also, add a reference to this section here https://github.com/slimbootloader/documentation/pull/74/files#diff-7878704aff6f8f291ef221a31c437735

File: source/security/key-management.rst [hunk 5]
```diff
@@ -39,3 +39,63 @@ Cryptographic keys used for signing |SPN| binaries should be securely managed.
...
+One could use either key id or complete path to signing keys while configuring in build scripts.
+
+.. note:: Signing tools support either KEY_ID corresponding to a component or complete path to private key.
+
+Test Keys Generation
+*********************
+
+Keys required for |SPN| can be generated using GenerateKeys.py available
+at BootloaderCorePkg/Tools/. The key generation process could be **one
```
Comment (rprangar) [inline source/security/key-management.rst:76]: the key generation process is a **one time...

File: source/security/key-management.rst [hunk 6]
```diff
@@ -39,3 +39,63 @@ Cryptographic keys used for signing |SPN| binaries should be securely managed.
...
+time process** for specific project. Use same set of keys for signing
+and verification operations for a specific project when generating
+firmware capsule update image, cfgdata stitch, Container image and
+others. Update operations would fail incase different keys are used
```
Comment (rprangar) [inline source/security/key-management.rst:80]: Maybe use the work "verification" instead of update?

File: source/security/key-management.rst [hunk 7]
```diff
@@ -39,3 +39,63 @@ Cryptographic keys used for signing |SPN| binaries should be securely managed.
...
+
+
+Build Environment Configuration for Key ID usage
+************************************************
+
+Key directory to be used can be specified using an environment variable.
+
+Set env variable “SBL_KEY_DIR” to keys directory generated using
+GenerateKeys.py or similar methods. This env variable need to be to set before running SBL
+build command. Also, set environment vaiable before executing tools in standalone mode as
```
Comment (rprangar) [inline source/security/key-management.rst:93]: typo: vaiable -> variable

File: source/security/key-management.rst (hunk 4 above)
Comment (lsubash) [reply source/security/key-management.rst:47]: Done

File: source/developer-guides/build-system.rst (hunk 1 above)
Comment (lsubash) [reply source/developer-guides/build-system.rst]: Updated in build-host-setup section

File: source/developer-guides/build-system.rst (hunk 3 above)
Comment (lsubash) [reply source/developer-guides/build-system.rst]: "Test" removed.

File: source/security/key-management.rst (hunk 4 above)
Comment (lsubash) [reply source/security/key-management.rst:47]: Updated as recommended

File: source/security/key-management.rst (hunk 5 above)
Comment (lsubash) [reply source/security/key-management.rst]: Done

File: source/security/key-management.rst (hunk 6 above)
Comment (lsubash) [reply source/security/key-management.rst]: Updated text.

**Pull request merged**